### PR TITLE
[opentelemetry-cpp] Update to 1.6.0 and fix abseil[cxx17]

### LINKED ISSUE #26854: [opentelemetry-cpp] update to 1.6.0
Library name: opentelemetry-cpp

New version number: 1.6.0

Using `abseil[cxx17]` is also broken. I already prepared a branch with the necessary changes, will open a pr soon.

### PR DESCRIPTION
**Pull request by Tradias**
**Describe the pull request**

- #### What does your PR fix?
  Fixes https://github.com/microsoft/vcpkg/issues/26854

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes

### CODE REVIEW

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JonLiu1993** [comment]
All features are tested successfully in the x64-windows-static triplet.

**Comment by vicroms** [comment]
Please fix merge conflicts

File: versions/o-/opentelemetry-cpp.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
   "versions": [
     {
-      "git-tree": "49b229180a92d7c0148fd1c28a691a76c59aea51",
-      "version-semver": "1.5.0",
-      "port-version": 1
+      "git-tree": "b85f6215a2936a5a21c752b11e16fc7250a102b2",
```
Comment (Thomas1664) [inline versions/o-/opentelemetry-cpp.json:4]: The merge conflict hasn't been fixed properly

Comment (Thomas1664) [reply versions/o-/opentelemetry-cpp.json:4]: I'm NOT a member of the vcpkg team. Therefore, I can't edit your PR. I suggest that you revert the changes to this file and to `baseline.json` and run `vcpkg x-add-version opentelemetry-cpp`

Comment (Tradias) [reply versions/o-/opentelemetry-cpp.json:4]: Is it correct now? I re-added 1.5.0p1

Comment (Thomas1664) [reply versions/o-/opentelemetry-cpp.json:4]: Yes, this is correct

File: ports/opentelemetry-cpp/support_absl_cxx17.patch [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+@@ -305,7 +305,11 @@ if(WITH_OTLP)
+                                                        CACHE{WITH_OTLP_HTTP}))
+     find_package(CURL)
+   endif()
+-
++  if (ABSL_USE_CXX17)
++    message(STATUS "Found absl uses CXX17, enable CXX17 feature.")
++    set(CMAKE_CXX_STANDARD 17)
++    set(WITH_STL ON)
++  endif()
```
[the remark this answers is not in the document]
Comment (Tradias) [reply ports/opentelemetry-cpp/support_absl_cxx17.patch:14]: This is an unfortunate side-effect of how vcpkg chooses to handle abseil[cxx17]. I wish they would just [patch abseil](https://github.com/abseil/abseil-cpp/blob/7f3c0d781134d25b24e0b55346482a2d366bf59b/CMake/AbseilHelpers.cmake#L303) such that it propagates `cxx_std_17`. That would make this if condition redundant for all CMake ports that depend directly or transitively on abseil, at least.

**Comment by Tradias** [comment]
Is there anything that blocks this pull request?

**Comment by JonLiu1993** [comment]
@vicroms, Could you please review again?

**Pull request merged**